jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -151,59 +140,63 @@ else if (urlString.startsWith("jar:"))
      */
     public static Resource newResource(String resource) throws IOException
     {
-        return newResource(resource, __defaultUseCaches);
+        URI uri = URI.create(resource);
+
+        // If the URI has no scheme, we consider the string actually was a path.
+        if (uri.getScheme() == null)
```

Comment:
To my surprise, that _is_ interpreted as a `URI`. I'm going to modify `Resource.newResource(String)` to try to interpret the string as a path first.

Reply:
Previously we tried URL first and only if that was Malformed did we try a Path.  So trying a URI first feels like less of a change.
Luckily `\` is not a valid character for URI, so that approach will still work.

Reply:
you also have paths like `//machine/share/directory/file.txt` on windows.
(which can also show up as `\\machine\share\directory\file.txt`)

Don't forget `file://c:/path/to/file.txt` which has 2 `:` (colon) symbols.

Then there's also (on windows) `c:path/relative/to/cwd/to/resource.txt` (note the lack of starting `/` after the drive symbol.

Reply:
There are enough gotchas to make me think `Resource.newResource(String)` is a bad API and should be removed to force the caller to specify if the argument is a path or a URI.

This string interpretation definitely has the potential to generate security issues.

Reply:
I  agree.

But we still need to support configuration from jetty-home/jetty-base, modules, ini properties, jetty xml, etc...

Reply:
Hmmmm it has worked well enough for over a decade and had made configuring the server flexible and portable.

We keep it with current behaviors, document the limitations and don't use it for serving resources, only for configuring the server.